### DISCUSSION TITLE
기본 폴더에 대한 논의

### DISCUSSION THREAD

**Discussion by Gyaak** [Kernel360/F2-BAGUNI#119] [개발 토의]
# 기본 폴더에 대한 논의
## 사용자가 회원가입하면 기본적으로 생기는 폴더가 2개 존재
1. 미분류폴더 - UNCLASSIFIED
2. 휴지통 - RECYCLE_BIN

두개의 기본 폴더는 회원가입 시점에 자동으로 생성되고, folderId를 가지게 됩니다.
저는 다음 두가지 방법을 생각해봤습니다.

## 1. 실제 folderId를 직접 사용
특정 사용자의  미분류폴더와 휴지통의 고유 folderId를 받는 api를 사용하여 기본 폴더의 folderId를 받을 수 있습니다.
```
// (주의)예시 api는 예시이므로 추후 변경될 수 있음

// 예시 api : /api/folder/unclassified
{
  "id" : "로그인된 유저의 미분류폴더 id"
}

// 예시 api : /api/folder/recycle_bin
{
  "id" : "로그인된 유저의 휴지통 id"
}
```

위 api를 활용하여 미분류폴더와 휴지통의 folderId를 획득하여 사용

## 2. 특수 id를 사용 ( 미분류폴더 : -1 , 휴지통 : -2 )
요청에 folderId를 -1 , -2로 보내면
server에서 자동으로 적절한 기본폴더로 인식

ex) pickId : 3 인 픽을 folderId : -1 로 이동
-> pickId : 3 인 픽을 미분류 폴더로 이동

---

@kimminkyeu @sangwonsheep @enigsuss @dmdgpdi
급한건 아니니 10/5(토)까지 확인 후 **_코멘트 혹은 이모지_** 남겨주세요

**Comment by kimminkyeu** [comment]
1번 방식이 파일 시스템 경로 접근과 유사해서 직관적인 것 같아요.

- `GET /api/folder/unclassified` --> 폴더 id + 미분류된 픽 리스트 반환
- `GET /api/folder/recycle_bin` --> 폴더 id + 휴지통에 있는 `폴더`|`픽` 리스트 반환
- `GET /api/folder/` --> 최상단에 있는 `폴더`|`픽` 리스트 반환

1번에 한표 던집니다.
